addons/bus/controllers/main.py

```diff
@@ -1,39 +1,12 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import json
-import logging
 
-from odoo import exceptions, _
 from odoo.http import Controller, request, route
-from odoo.addons.bus.models.bus import dispatch
-
+from odoo.addons.bus.models.bus import channel_with_db, dispatch
+from odoo.addons.bus.websocket import Websocket
 
 class BusController(Controller):
-
-    # override to add channels
-    def _poll(self, dbname, channels, last, options):
-        channels = list(channels)  # do not alter original list
-        channels.append('broadcast')
-        # update the user presence
-        if request.session.uid and 'bus_inactivity' in options:
-            request.env['bus.presence'].update(inactivity_period=options.get('bus_inactivity'), identity_field='user_id', identity_value=request.session.uid)
-        request.env.cr.close()
-        return dispatch.poll(dbname, channels, last, options)
-
-    @route('/longpolling/poll', type="json", auth="public", cors="*")
-    def poll(self, channels, last, options=None):
-        if options is None:
-            options = {}
-        if not dispatch:
-            raise Exception("bus.Bus unavailable")
-        if [c for c in channels if not isinstance(c, str)]:
-            raise Exception("bus.Bus only string channels are allowed.")
-        if request.registry.in_test_mode():
-            exc = exceptions.UserError(_("bus.Bus not available in test mode"))
-            exc.loglevel = logging.INFO
-            raise exc
-        return self._poll(request.db, channels, last, options)
-
     @route('/longpolling/im_status', type="json", auth="user")
```

Comment:
What's the plan w.r.t `/longpolling/` prefixed routes? Are you planning on moving them all in websocket space? I've just skimmed through the diff and it looks like you're dropping the gevent worker completely. So I would think we should drop the `/longpolling/` namespace as well to avoid confusion (and avoid breaking existing proxy setups).

Reply:
I see nothing wrong in droping the `/longpolling` namespace. As for the gevent worker, it will still be used but a little configuration will be needed to redirect `/websocket` routes instead of `/longpolling` ones.

Reply:
Hello @julienlegros, the gevent worker is still being used, it is used to serve HTTP Upgrade requests (`/websocket` route) and the websockets themselves.